torch/_dynamo/output_graph.py

```diff
@@ -453,7 +461,12 @@ def call_user_compiler(self, gm):
                 else ""
             )
             _step_logger()(logging.INFO, f"calling compiler function {name}")
-            compiled_fn = self.compiler_fn(gm, self.example_inputs())
+            if "shape_env" in signature(self.compiler_fn).parameters.keys():
```

Comment:
do we have precedent for this kind of argument sniffing? cc @Chillee IIRC last time we changed the compiler calling convention we did it by adding a tag to the function (for boxed args)